pkg/systemd/terminal.jsx

```diff
@@ -95,6 +136,42 @@ const _ = cockpit.gettext;
             document.cookie = cookie;
         }
 
+        onModal(){
+            this.setState({ changePathBusy: false });
+            this.setState(prevState => ({ channel: this.createChannel(prevState.user, cockpit.location.options.path)}));
+        }
+
+        async onNavigate(){
+            // Clear old path errors
+            this.setState({ pathError: null })
+
+            // If there's no path to change to, then we're done here
+            if (!cockpit.location.options.path) {
```

Comment:
... or if it's the same as the current one? This could happen realistically, I think.

If the current path is e.g. /var, and you call the terminal without a path, then I agree it's reasonable to not do anything.

Reply:
I think onNavigate() doesn't trigger unless the path is changed, since the cockpit event listener that triggers onNavigate() is locationchanged?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event (on which `locationchanged` is based) isn't super clear on this. Let's add an integration test that b.location.go()'s to the same path with a busy terminal, and make sure it doesn't trigger an alert.